app/scripts/controllers/detect-tokens.js

```diff
@@ -32,6 +33,7 @@ export default class DetectTokensController {
     this.network = network;
     this.keyringMemStore = keyringMemStore;
     this.tokenList = tokenList;
+    this.useTokenDetection = this.preferences?.store.getState().useTokenDetection;
```

Comment:
@NiranjanaBinoy I'm not clear on why this change was necessary?

Reply:
it is not, I was trying another fix at first then switched to this one, this entry was made for the other attempt. I will remove it now.